### PR TITLE
Invalidiate schema and friends caches when shema prop is updated

### PR DESCRIPTION
**Pull request by thetrompf**

### CODE REVIEW

File: src/form.jsx [hunk 1]
```diff
@@ -54,13 +54,13 @@ Shift.Form = ShiftForm = React.createClass({
...
 			fieldInFocus: null,
-			fieldErrors: this.getEmptyFieldErrors(),
+			fieldErrors: this.getEmptyFieldErrors(this.props),
 			submittedOnce: false,
 			presenterValues: this.props.initialValue || {}
 		};
 	},
 
-	getEmptyFieldErrors: function(){
+	getEmptyFieldErrors: function(props){
```
Comment (bondo) [inline src/form.jsx]: Why pass in `props`? It's not used...

File: src/form.jsx [hunk 2]
```diff
@@ -54,13 +54,13 @@ Shift.Form = ShiftForm = React.createClass({
...
 		var result = {};
 
 		for(var key in this.props.schema){
```
Comment (bondo) [inline src/form.jsx]: It's supposed to be used here?

Comment (kastermester) [inline src/form.jsx]: @thetrompf  can you do another pull request on this fixing this error? Sorry I didn't catch it the first time around.
